src/domain/registration/form/RegistrationForm.tsx

```diff
@@ -194,65 +239,81 @@ const RegistrationForm: FunctionComponent = () => {
                       'registration.form.guardian.email.input.placeholder'
                     )}
+                    invalid={
+                      getIn(touched, 'guardian.email') &&
+                      getIn(errors, 'guardian.email')
+                    }
+                    helperText={t(getIn(errors, 'guardian.email'))}
                   />
-                  <EnhancedInputField
+                  <Field
+                    as={TextInput}
                     id="guardian.phoneNumber"
                     name="guardian.phoneNumber"
                     required={true}
                     label={t(
                       'registration.form.guardian.phoneNumber.input.label'
                     )}
-                    component={InputField}
                     placeholder={t(
                       'registration.form.guardian.phoneNumber.input.placeholder'
                     )}
+                    invalid={
+                      getIn(touched, 'guardian.phoneNumber') &&
+                      getIn(errors, 'guardian.phoneNumber')
+                    }
+                    helperText={t(getIn(errors, 'guardian.phoneNumber'))}
                   />
                   <div className={styles.guardianName}>
-                    <EnhancedInputField
-                      type="text"
+                    <Field
+                      as={TextInput}
                       required={true}
                       id="guardian.firstName"
                       name="guardian.firstName"
                       label={t(
                         'registration.form.guardian.firstName.input.label'
                       )}
-                      component={InputField}
                       placeholder={t(
                         'registration.form.guardian.firstName.input.placeholder'
                       )}
+                      invalid={
+                        getIn(touched, 'guardian.firstName') &&
+                        getIn(errors, 'guardian.firstName')
+                      }
+                      helperText={t(getIn(errors, 'guardian.firstName'))}
                     />
-                    <EnhancedInputField
-                      type="text"
+                    <Field
+                      as={TextInput}
                       required={true}
                       id="guardian.lastName"
                       name="guardian.lastName"
                       label={t(
                         'registration.form.guardian.lastName.input.label'
                       )}
-                      component={InputField}
                       placeholder={t(
                         'registration.form.guardian.lastName.input.placeholder'
                       )}
+                      invalid={
+                        getIn(touched, 'guardian.lastName') &&
+                        getIn(errors, 'guardian.lastName')
+                      }
+                      helperText={t(getIn(errors, 'guardian.lastName'))}
```

Comment:
If you create a wrapper for TextInput, you would most likely be able to provide default error logic for the component. That way you don't have to repeat yourself as much when setting invalid and helperText.

I made a quick and dirty version of this for the feedback component. I used a hook to be able to share the error logic between different kind of form components:
https://github.com/City-of-Helsinki/kuva-feedback-component/blob/develop/src/common/components/formikWrappers/Input.tsx
https://github.com/City-of-Helsinki/kuva-feedback-component/blob/develop/src/common/hooks/useField.tsx

There's only a single form in the app, so it didn't need to be that flexible of an implementation.

Reply:
Great suggestion! 

This is something that would probably come to mind if I was reviewing someone else's PR :-)

Reply:
This wrapper is now done.